File: express-rust/src/http/method.rs
pub struct Method(Inner);

enum Inner {
    Options,
    Get,
    Post,
    Put,
    Patch,
    Delete,
}

impl Method {
    // OPTIONS
    pub const OPTIONS: Method = Method(Options);

    // GET
    pub const GET: Method = Method(Get);

    // POST
    pub const POST: Method = Method(Post);

    // PUT
    pub const PUT: Method = Method(Put);

    // PATCH
    pub const PATCH: Method = Method(Patch);

    // DELETE
    pub const DELETE: Method = Method(Delete);
}

File: http-server-with-typescript/app/http/method.ts
export enum Method {
  GET = "GET",
  POST = "POST",
  PUT = "PUT",
  DELETE = "DELETE",
}

export type MethodType = Method.GET | Method.POST | Method.PUT | Method.DELETE;

export const getMethod = (methodString: string): MethodType => {
  switch (methodString) {
    case "GET":
    default:
      return Method.GET;
    case "POST":
      return Method.POST;
    case "PUT":
      return Method.PUT;
    case "DELETE":
      return Method.DELETE;
  }
};
